### ISSUE TITLE
Add high resistance and high voltage components

### ISSUE THREAD

**Issue by ariel-phet** [phetsims/circuit-construction-kit-dc#29]
A twitter user was wondering if there was a way to go above 100 ohms on resistance (not possible in Java sim)

Perhaps we could have a "low" resistance resistor and a "high" resistance resistor on the lab screen. It does seem helpful to have the ability to go above 100 ohms.

**Comment by arouinfar** [comment]
@ariel-phet I think it's definitely something we should consider. I've been combing through the relevant Unfuddle tickets, and we've had a similar suggestion before.

Unfuddle Ticket [#3607](https://phet.unfuddle.com/a#/projects/9404/tickets/by_number/3607?cycle=true)
> I am a Physics teacher at Hoover High School in San Diego, CA.  The students in my introductory classes are studying electric circuits right now. Right now, my student teacher and I are currently putting together an activity using the Circuit Construction Kit where students build a simple circuit with resistors representing 11 lamps in various rooms of a person's home.  In order to make the simulation reasonably realistic I would like to insert resistor values of 250 ohms,  1000 ohms and 1600 ohms to represent incandescent bulbs, CFL bulbs and LED bulbs respectively.  I would also like to set the voltage source at 120 volts DC.  I realize that a real home will use AC, but for my purposes, I will be using DC only, to keep it simple.
The problem I am facing now is that the simulation only seems to permit a maximum resistance of 100 ohms. Is there any way I can easily customize those values to be 250, 1000 and 1600 ohms so that the students can use reasonably realistic values in their measurements and calculations.

**Comment by arouinfar** [comment]
Discussed in 3/23/17 design meeting. 

We decided to create three additional components - high V battery, high R bulb, high R resistor. Self-assigning to mock up these new components.

**Comment by arouinfar** [comment]
Some options for high-R components. High-R resistors and batteries are 33% taller than the standard components. 
![image](https://cloud.githubusercontent.com/assets/8419308/25026835/bee8892e-2065-11e7-8246-bb87a209a6ae.png)

We'll also need to decide where these items belong in the carousel. Should they be grouped with standard version, or on a later page of the carousel?

**Comment by arouinfar** [comment]
Based on the design meeting discussion, I've mocked up the high V/high R components. Before I send the assets off to @samreid, can you take a look @ariel-phet? Any suggestions?

![image](https://cloud.githubusercontent.com/assets/8419308/25597284/243e9290-2e8b-11e7-9b74-54bcb82058d7.png)

**Comment by ariel-phet** [comment]
@arouinfar I think the high voltage battery looks good. I might make the stroke on the resistor symbol fatter. Maybe 50% more? I also wonder if it would look good in a color that is not the "schematic" color.

**Comment by arouinfar** [comment]
Thanks @ariel-phet! I cleaned up the resistor symbol a bit, increased the stroke width, and played with color. Darker colors will have better contrast against copper, and I think navy looks pretty good without being too flashy.

![image](https://cloud.githubusercontent.com/assets/8419308/25872841/6259cbbe-34c9-11e7-8073-90bc9e88cea2.png)

**Comment by ariel-phet** [comment]
@arouinfar, I like it, and agree on the navy

**Comment by arouinfar** [comment]
Thanks @ariel-phet! I'll send the assets to @samreid.

**Comment by arouinfar** [comment]
Assets have been emailed to @samreid.

**Comment by samreid** [comment]
@arouinfar which screens will feature these high-value components?

**Comment by samreid** [comment]
We may need high voltage components on the grab bag screen to get currents running through the high resistance elements?

**Comment by samreid** [comment]
What are the ranges and default values for these new components?

**Comment by arouinfar** [comment]
@samreid here's what we originally discussed:
* Lab screen only, 2nd page of carousel between regular components and grab bag items.
* High R resistor range: 100-10^4 ohms, default 1000 ohms
* High V battery range: 100-10^5 V, default 1000 V

> We may need high voltage components on the grab bag screen to get currents running through the high resistance elements?

I think the goal for the Explore screen is for students to discover the difference between conductors and resistors. I think having high V components on the Explore screen may muddy the waters a bit, because students will be able to get a measurable current through the resistors.

**Comment by samreid** [comment]
@arouinfar what labels do you want to show under these new components?  Something like:

* High Voltage Battery
* High Resistance Resistor
* High Resistance Light Bulb

Or is that too wordy and we just want to call them "Battery" etc and let the icons do the talking?

**Comment by arouinfar** [comment]
@samreid I don't think the label needs to explicitly include high V/R. Let's just call them "Battery", "Resistor", and "Light Bulb".

**Comment by samreid** [comment]
The high voltage battery triangle overlaps the default electron location.

![image](https://user-images.githubusercontent.com/679486/26888516-7e458326-4b68-11e7-9cd4-5a19a7aef59f.png)

@arouinfar what do you think about nudging the triangle to the right so it is centered between those electrons?

**Comment by samreid** [comment]
Should we do something similar for the resistor? (put the icon 1/3 across instead of 1/2 way across?)

**Comment by samreid** [comment]
> The high voltage battery triangle overlaps the default electron location.

Or maybe it is OK because it is shown in the toolbox with no electrons:

![image](https://user-images.githubusercontent.com/679486/26890463-d4336518-4b6e-11e7-9b22-8c99ab3cfd75.png)

Bumping the triangle may still may be a good improvement though, just jotting down my thoughts.

**Comment by arouinfar** [comment]
@samreid I could center the caution symbol within the black part of the battery, but I think shifting the resistor symbol looks unbalanced.
![image](https://user-images.githubusercontent.com/8419308/26893041-68f063c4-4b77-11e7-9c6e-22c962a9152c.png)

I don't see the electron occlusion as particularly harmful because the symbol is clearly visible in the carousel, and once a circuit is built the electrons will be moving.

**Comment by samreid** [comment]
@arouinfar for the battery, what increment/decrement do you want the tweaker buttons to take?  If I made no code changes it would remain at 0.5V.

**Comment by samreid** [comment]
> @samreid I could center the caution symbol within the black part of the battery, but I think shifting the resistor symbol looks unbalanced.

Sounds good, I recommend to move the symbol for the battery but leave it centered for the resistor.  Can you send new artwork for the battery?

**Comment by samreid** [comment]
Just to clarify, we are not showing bands on the high resistance resistors? Is that outside of the learning goals?

**Comment by samreid** [comment]
With this range and the rendering of the slider, the slider looks like it is all the way to the left (even though it can be dragged all the way down to 100)

![image](https://user-images.githubusercontent.com/679486/26903457-83368074-4b9a-11e7-9e62-06c8f4da296a.png)

@arouinfar how do you want to adjust this so students have a cue they can reduce below 1000?

**Comment by samreid** [comment]
@arouinfar Do you want a blank spot in the high components carousel page?

![image](https://user-images.githubusercontent.com/679486/26904538-9d4727a2-4b9f-11e7-84f8-f92f24e02b0b.png)

**Comment by samreid** [comment]
These specs https://github.com/phetsims/circuit-construction-kit-dc/issues/29#issuecomment-303919729 do not include the range or default for light bulb resistance.  I'll go with the same values as the resistor, but @arouinfar please correct me if this should be different.

**Comment by samreid** [comment]
In the above commit, I eliminated the blank spot in the carousel.

**Comment by samreid** [comment]
Initial revision is here:
http://www.colorado.edu/physics/phet/dev/html/circuit-construction-kit-dc/1.0.0-dev.54/circuit-construction-kit-dc_en.html

@arouinfar can you please review?

I'll move my questions above to new issues so they don't get lost.  They will be linked below shortly.

**Comment by arouinfar** [comment]
> @arouinfar for the battery, what increment/decrement do you want the tweaker buttons to take? If I made no code changes it would remain at 0.5V.

Let's set the tweakers to 100.

> Sounds good, I recommend to move the symbol for the battery but leave it centered for the resistor. Can you send new artwork for the battery?

New artwork has been sent!

> Just to clarify, we are not showing bands on the high resistance resistors? Is that outside of the learning goals?

We want the resistors to be visually distinct from one other, and learning goals surrounding resistor bands could be satisfied with the standard resistor.

> @arouinfar how do you want to adjust this so students have a cue they can reduce below 1000?

Looks like the slider ranges for high R and high V have been swapped. On master, R goes all they way up to 100,000 ohms, which matches the hand. We had decided the upper limit would be 10^4, or 10,000 ohms. The high V battery should match Java, which goes up to 100,000 V, but is currently maxing out at 10,000 V. Once you swap the ranges, the Resistance slider should no longer have a problem at the default 1000 ohms -- but the Voltage slider will.

To better cue the slider range, let's increase the default voltage to 10,000. (Using screenshot of Resistance slider as a stand in because it has the desired range.)
![image](https://user-images.githubusercontent.com/8419308/26905313-ed7b2a76-4ba3-11e7-839c-aa5baf92edba.png)


> @arouinfar Do you want a blank spot in the high components carousel page?

I don't particularly like the blank spot, so I'm glad you've gone ahead and eliminated it.

> I'll go with the same values as the resistor, but @arouinfar please correct me if this should be different.

The lightbulb range/default should match the resistor.

**Comment by arouinfar** [comment]
Looks like the conversation didn't refresh while I was commented, so I missed https://github.com/phetsims/circuit-construction-kit-dc/issues/29#issuecomment-306951528.  I took a look at dev.54 and things look pretty good, save for the swapped voltage and resistance ranges I referred to in https://github.com/phetsims/circuit-construction-kit-dc/issues/29#issuecomment-306951927.

**Comment by samreid** [comment]
Battery triangle icon looks much better, thanks!

![image](https://user-images.githubusercontent.com/679486/26909704-b10eaf10-4bbf-11e7-8945-4f08ad7a8963.png)

**Comment by samreid** [comment]
@arouinfar I published a new version here:
http://www.colorado.edu/physics/phet/dev/html/circuit-construction-kit-dc/1.0.0-dev.55/circuit-construction-kit-dc_en.html

Can you please review it?

**Comment by arouinfar** [comment]
@samreid everything looks good, but I want to leave this this open and self-assigned for a deeper review to see if there are any side-effects of the high V/high R components that we'll need to account for in the design.

**Comment by arouinfar** [comment]
@samreid I'm going to go ahead and close this issue, as the high V/high R components have been successfully implemented. I'll add my concerns to the meeting notes for today's discussion.
